Library/Homebrew/download_strategy.rb

```diff
@@ -388,6 +388,25 @@ def _fetch
   end
 end
 
+# Will convert "https://downloads.sourceforge.net/project/..." URLs to mirrorservice.org
+# In case environment variable HOMEBREW_NO_INSECURE_REDIRECT is set, so the Sourceforge is not ultimately broken
+class CurlSourceForgeSecureMirrorDownloadStrategy < CurlDownloadStrategy
+  def fetch
+    unless ENV["HOMEBREW_NO_INSECURE_REDIRECT"].nil?
+      ohai "Sourceforge HTTPS: #{@url}"
+      @url = https_mirror
+      ohai "=> converted to #{@url}"
+    end
+    super
+  end
+
+  def https_mirror
+     @url.gsub(%r{(^https?://)(?:.+)(.net/project/)(.+)}) do
+       "#{$1}www.mirrorservice.org/sites/download.sourceforge.net/pub/sourceforge/#{$3[0]}/#{$3[0,2]}/#{$3}"
```

Comment:
I observe difference between `download.sourceforge.net`, `downloads.sourceforge.net` and `dl.sourceforge.net`, I think they need to be handled separately, see the comment https://github.com/Homebrew/homebrew/pull/42955#issuecomment-132358467